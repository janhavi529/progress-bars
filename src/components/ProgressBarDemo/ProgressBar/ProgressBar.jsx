import React from 'react';
import PropTypes from 'prop-types';
import '../../../styles/scss/components/_progress-bar.scss';

const progressBar = ({ value, limit }) => {
  // Dynamically set the width and color of progress bar.
  const progressStyles = { 
    width: (value < 0 ? 0 : (value > limit || value > 100 ? 100 : value)) + '%',
    backgroundColor: value > limit ? 'lightCoral' : 'lightGreen'
  };

  return (
    <div className="bar-progress">
      <span className="bar-progress-percentage">{value}%</span>
      <div className="bar-progress-tracker" style={progressStyles}></div>
    </div>
  );
}

progressBar.propTypes = {
  value: PropTypes.number.isRequired,
  limit: PropTypes.number.isRequired,
};

export default progressBar;