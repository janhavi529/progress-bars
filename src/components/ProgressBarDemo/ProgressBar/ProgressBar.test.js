import React from 'react';
import { shallow } from 'enzyme';
import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
    const props = {
        value: 45,
        limit: 150
    };

    it('Renders ProgressBar with default props', () => {
        const wrapper = shallow(<ProgressBar {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('ProgressBar render when value is greater than limit', () => {
        const props1 = {
            value: 175,
            limit: 150
        };
        const wrapper = shallow(<ProgressBar {...props1}/>);

        expect(wrapper.find('.bar-progress-tracker').prop('style')).toHaveProperty('backgroundColor', 'lightCoral');
        expect(wrapper.find('.bar-progress-percentage').text()).toEqual('175%');
        expect(wrapper).toMatchSnapshot();
    });

    it('ProgressBar render when value is less than 0', () => {
        const props = {
            value: -45,
            limit: 150
        };
        const wrapper = shallow(<ProgressBar {...props}/>);
        expect(wrapper.find('.bar-progress-tracker').prop('style')).toHaveProperty('backgroundColor', 'lightGreen');
        expect(wrapper.find('.bar-progress-percentage').text()).toEqual('-45%');
        expect(wrapper).toMatchSnapshot();
    });
});