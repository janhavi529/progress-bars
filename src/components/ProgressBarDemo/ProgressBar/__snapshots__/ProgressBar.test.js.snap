// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar ProgressBar render when value is greater than limit 1`] = `ShallowWrapper {}`;

exports[`ProgressBar ProgressBar render when value is less than 0 1`] = `ShallowWrapper {}`;

exports[`ProgressBar Renders ProgressBar with default props 1`] = `ShallowWrapper {}`;
